### PR TITLE
Fix missing expansion files in clientset release 1.4

### PR DESCRIPTION
**Pull request by caesarxuchao**
**What this PR does / why we need it**:
Fix missing expansion files in clientset release 1.4;
Update the client-gen tool to automatically update generated_expansion.go;
Update the readme to document the best practice of using client-gen.

**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #

https://github.com/kubernetes/kubernetes/issues/29698#issuecomment-241315798

**Special notes for your reviewer**:

**Release note**:

``` release-note
```

cc @lavalamp @krousey @mbohlool

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31470)

### CODE REVIEW

File: docs/devel/generating-clientset.md [hunk 1]
```diff
@@ -62,12 +62,16 @@ will generate a clientset named "my_release" which includes clients for api/v1
...
-adds additional methods to Pod's client. As a convention, we put the expansion
-interface and its methods in file ${TYPE}_expansion.go.
+- ***Adding expansion methods***: client-gen only generates the common methods,
+  such as `Create()` and `Delete()`. You can manually add additional methods
+  through the expansion interface. For example, this
+  [file](../../pkg/client/clientset_generated/release_1_4/typed/core/v1/pod_expansion.go)
+  adds additional methods to Pod's client. As a convention, we put the expansion
+  interface and its methods in file ${TYPE}_expansion.go. In most cases, you
+  don't want to remove existing expansion files. So to make life easier,
+  instead of creating a new clientset from scratch, ***you can copy and rename an
```
Comment (caesarxuchao) [inline docs/devel/generating-clientset.md]: @smarterclayton FYI, I remember you once complained about missing expansion files when generating clientset.

**Comment by nikhiljindal** [comment]
Thanks, these changes LGTM.
Do you know why were these expansion files missing in the initial copy?
We should know so that we can make sure that it does not happen again.

**Comment by caesarxuchao** [comment]
I guess I forgot to call `git add` after manually copying those expansion files. Following the new best practices in docs/devel/generating-clientset.md should prevent this from happening again.

**Comment by caesarxuchao** [comment]
@nikhiljindal can you add the milestone? Large part of this PR is fixing a bug in 1.4 clientset, so I think it's worth the milestone.

**Comment by lavalamp** [comment]
Yeah we definitely need this for 1.4.

**Comment by caesarxuchao** [comment]
Run client-gen for internalclientset and federation clientset to satisfy the verification script. generated_expansion.go hadn't been automatically updated for those clientsets for a long time.
Re-add the lgtm.

**Comment by caesarxuchao** [comment]
Fixed the subjectreview expansion file in the internalversion clientset.

**Comment by lavalamp** [comment]
note to release czar if you're looking: this is a low-risk bug fix, 1.4 versioned client is quite broken without this.

**Comment by caesarxuchao** [comment]
@k8s-bot node e2e test this, issue #IGNORE

**Pull request merged**